autosklearn/data/target_validator.py

```diff
@@ -0,0 +1,385 @@
+                y_train = y_train + y_test
+            elif isinstance(y_train, np.ndarray):
+                y_train = np.concatenate((y_train, y_test))
+
+        ndim = len(np.shape(y_train))
+        if ndim == 1 or (ndim > 1 and np.shape(y_train)[1] == 1):
+            # The label encoder makes sure data is, and remains
+            # 1 dimensional
+            self.encoder = preprocessing.OrdinalEncoder(handle_unknown='use_encoded_value',
+                                                        unknown_value=-1)
+        else:
+            # We should not reach this if statement as we check for type of targets before
+            raise ValueError("Multi-dimensional classification is not yet supported. "
+                             "Encoding multidimensional data converts multiple columns "
+                             "to a 1 dimensional encoding. Data involved = {}/{}".format(
+                                np.shape(y_train),
+                                self.type_of_target
+                             ))
+
+        # Mypy redefinition
+        assert self.encoder is not None
+
+        # remove ravel warning from pandas Series
+        if ndim > 1:
+            self.encoder.fit(y_train)
+        else:
+            if hasattr(y_train, 'iloc'):
+                y_train = typing.cast(pd.DataFrame, y_train)
+                self.encoder.fit(y_train.to_numpy().reshape(-1, 1))
+            else:
+                self.encoder.fit(np.array(y_train).reshape(-1, 1))
+
+        # we leave objects unchanged, so no need to store dtype in this case
+        if hasattr(y_train, 'dtype'):
+            # Series and numpy arrays are checked here
+            # Cast is as numpy for mypy checks
+            y_train = typing.cast(np.ndarray, y_train)
+            if is_numeric_dtype(y_train.dtype):
+                self.dtype = y_train.dtype
+        elif hasattr(y_train, 'dtypes') and is_numeric_dtype(typing.cast(pd.DataFrame,
+                                                                         y_train).dtypes[0]):
+            # This case is for pandas array with a single column
+            y_train = typing.cast(pd.DataFrame, y_train)
+            self.dtype = y_train.dtypes[0]
+
+        return self
+
+    def transform(
+        self,
+        y: typing.Union[SUPPORTED_TARGET_TYPES],
+    ) -> np.ndarray:
+        """
+        Validates and fit a categorical encoder (if needed) to the features.
+        The supported data types are List, numpy arrays and pandas DataFrames.
+
+        Parameters
+        ----------
+            y: SUPPORTED_TARGET_TYPES
+                A set of targets that are going to be encoded if the current task
+                is classification
+        Return
+        ------
+            np.ndarray:
+                The transformed array
+        """
+        if not self._is_fitted:
+            raise NotFittedError("Cannot call transform on a validator that is not fitted")
+
```

Comment:
I believe there is a `self._check_data(X)` missing here?

Reply:
Thanks, I have added this so we can check for test data in case it is provided not through the fit method